tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -422,6 +426,174 @@ def verify_failure_received():
         finally:
             lambda_client.delete_event_source_mapping(UUID=event_source_mapping_uuid)
 
+    @pytest.mark.aws_validated
+    @pytest.mark.parametrize(
+        "item_to_put1, item_to_put2, filter, calls",
+        [
+            # Localstack behaviour: triggered twice
+            (
+                {"id": {"S": "test123"}, "id2": {"S": "test42"}},
+                None,
+                None,
+                1,
+            ),
+            # Test with filter, and two times same entry
+            (
+                {"id": {"S": "test123"}, "id2": {"S": "test42"}},
+                None,
+                {"eventName": ["INSERT"]},
+                1,
+            ),
+            # Test with OR filter
+            (
+                {"id": {"S": "test123"}},
+                {"id": {"S": "test123"}, "id2": {"S": "42test"}},
+                {"eventName": ["INSERT", "MODIFY"]},
+                2,
+            ),
+            # Test with 2 filters (AND), and two times same entry (second time modified aka MODIFY eventName)
+            (
+                {"id": {"S": "test123"}},
+                {"id": {"S": "test123"}, "id2": {"S": "42test"}},
+                {"eventName": ["INSERT"], "eventSource": ["aws:dynamodb"]},
+                1,
+            ),
+            # Test exists filter
+            (
+                {"id": {"S": "test123"}},
+                {"id": {"S": "test1234"}, "presentKey": {"S": "test123"}},
+                {"dynamodb": {"NewImage": {"presentKey": [{"exists": False}]}}},
+                1,
+            ),
+            # numeric filters
+            # NOTE: numeric filters seem not to work with DynamoDB as the values are represented as string
+            # and it looks like that there is no conversion happening
+            # I leave the test here in case this changes in future.
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "123"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": {"N": [{"numeric": [">", 100]}]}}}},
+                0,
+            ),
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "100"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {
+                    "dynamodb": {
+                        "NewImage": {"numericFilter": {"N": [{"numeric": [">=", 100, "<", 200]}]}}
+                    }
+                },
+                0,
+            ),
+            # Prefix
+            (
+                {"id": {"S": "test123"}, "prefix": {"S": "us-1-testtest"}},
+                {"id": {"S": "test1234"}, "prefix": {"S": "testtest"}},
+                {"dynamodb": {"NewImage": {"prefix": {"S": [{"prefix": "us-1"}]}}}},
+                1,
+            ),
+        ],
+    )
+    def test_dynamo_db_insert_event(
```

Comment:
NIT: maybe you could rename this test to something more meaningful? like `test_dynamodb_event_filter`